bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudService.java

```diff
@@ -292,7 +298,9 @@ private String getSecret() {
         String newSecretString = "";
 
         if (!file.exists()) {
-            newSecretString = RandomStringUtils.randomAlphanumeric(20);
+            byte[] array = new byte[20];
+            new Random().nextBytes(array);
+            newSecretString = new String(array, StandardCharsets.UTF_8);
```

Comment:
This most likely won't generate an alphanumeric string. So this change isn't equivalent.

Reply:
Do you have an idea how it could be solved? Something like this

```java
UUID.randomUUID().toString().replaceAll("-", "");
```

Reply:
You could also do:
```suggestion
            byte[] array = new byte[10];
            new Random().nextBytes(array);
            newSecretString = HexUtils.bytesToHex(array);
```
That should produce a random enough alphanumeric string.

Reply:
Alright, but is it secure enough?

While touching it anyways we can ask @digitaldan for his opinion and if we maybe should increase the number of characters for the secret.

Reply:
If you want something truly equivalent to the original code, you could just make a utility method:
```
static final String AB = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
static SecureRandom rnd = new SecureRandom();

String randomString(int len){
   StringBuilder sb = new StringBuilder(len);
   for(int i = 0; i < len; i++)
      sb.append(AB.charAt(rnd.nextInt(AB.length())));
   return sb.toString();
}
```

Reply:
That sounds like the best option for me. Thanks for your help, I will incorporate it later.